WordPress/Classes/Services/BlogService.m

```diff
@@ -356,6 +356,21 @@ - (NSArray *)blogsForAllAccounts
     return [self blogsWithPredicate:nil];
 }
 
+- (NSDictionary *)blogsForAllAccountsById
+{
+    NSMutableDictionary *blogMap = [NSMutableDictionary dictionary];
```

Comment:
I don't recommend going this path, though.  One of the reasons is readability: in the snippet you propose above `@[]` should be `@{}` instead but is easy to miss.  The proposed snippet though will throw no warnings and crash during execution.

So basically:

```objective-c
NSMutableDictionary *dictionary = [@[] mutableCopy]; // no warning, hard to spot
NSMutableDictionary *dictionary = [NSMutableArray array]; // throws warning, easy to spot
```

From the benchmarking point of view there's also a slight advantage in using `[NSMutableDictionary dictionary];` as it's probably not performing any additional operations other than the basic allocation.  I admit this is negligible, although it's still a bonus.

For the following code:

```objective-c
- (void)benchmark
{
	NSDate *methodStart = [NSDate date];
	
	int counter = 65536;
	
	for (int i = 0; i < counter; i++) {
		//NSMutableDictionary* dictionary = [@{} mutableCopy];
		NSMutableDictionary* dictionary = [NSMutableDictionary dictionary];
	}
	
	NSDate *methodFinish = [NSDate date];
	NSTimeInterval executionTime = [methodFinish timeIntervalSinceDate:methodStart];
	NSLog(@"executionTime = %f", executionTime);
	
	exit(0);
}
```

`[NSMutableDictionary dictionary]` yields:
`2015-08-13 08:27:48.792 adasdasdasd[1253:23385] executionTime = 0.010310`

`[@{} mutableCopy]` yields:
`2015-08-13 08:28:39.583 adasdasdasd[1274:24177] executionTime = 0.014650`

Again, this is negligible although it sort of adds up to my argument above.  Just my 2 cents.